src/server/acl/acl_family.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+
+#include "server/acl/acl_family.h"
+
+#include "server/acl/acl_commands_def.h"
+#include "server/command_registry.h"
+#include "server/conn_context.h"
+
+namespace dfly::acl {
+
+constexpr uint32_t kList = acl::ADMIN | acl::SLOW | acl::DANGEROUS;
+
+static std::string AclToString(uint32_t acl_category) {
+  std::string tmp;
+
+  if (acl_category == acl::ALL) {
+    return "+@all";
+  }
+
+  if (acl_category == acl::NONE) {
+    return "+@none";
+  }
+
+  const std::string prefix = "+@";
+  const std::string postfix = " ";
+
+  for (uint32_t step = 0, cat = 0; cat != JSON; cat = 1ULL << ++step) {
+    if (acl_category & cat) {
+      tmp += prefix;
+      tmp += REVERSE_CATEGORY_INDEX_TABLE[step];
+      tmp += postfix;
+    }
+  }
+  tmp.erase(tmp.size());
+
+  return tmp;
+}
+
+void AclFamily::List(CmdArgList args, ConnectionContext* cntx) {
+  auto registry_with_lock = cntx->user_registry->GetRegistryWithLock();
+  const auto& registry = registry_with_lock.registry;
+
+  std::string buffer(255, ' ');
+  buffer.replace(0, 5, "user ");
+
+  auto resize_if_exceeds_size = [](std::string& buffer, size_t size) {
+    if (buffer.size() < size) {
+      buffer.resize(size);
+    }
+  };
+
+  auto step = [&resize_if_exceeds_size](std::string& buffer, size_t& size, size_t increment) {
+    size += increment;
+    resize_if_exceeds_size(buffer, size);
+  };
+
+  (*cntx)->StartArray(registry.size());
+  for (const auto& [username, user] : registry) {
```

Comment:
Not the biggest fun of this body, I will simplify once we get more things to print like subcommands. 

p.s. open for suggestions to simplify.

Reply:
I don't expect `ACL LIST` to be called a lot of times -- it should be used by admins maybe a few times.

What I was thinking about is that for 1000 users with an average of 200 characters we would need `200kb` roughly, The memory footprint would be low, but it would require allocating `1000` different strings I am not a big fun of this TBH, but I guess you are right this is a micro-optimization. It's not like this function will be called a lot of times....

Reply:
yes, but if you assemble the string from small parts with StrAppend() most will be covered by SSO